opencensus-migration/src/main/java/SpanCache.java

```diff
@@ -0,0 +1,45 @@
+import io.opencensus.implcore.trace.RecordEventsSpanImpl;
+import io.opencensus.trace.Span;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpanCache {
+  private static SpanCache spanCache;
+  private Map<Span, io.opentelemetry.trace.Span> ocToOt = new HashMap<>();
+  private Map<io.opentelemetry.trace.Span, Span> otToOc = new HashMap<>();
+
+  private SpanCache() {}
+
+  public static SpanCache getInstance() {
+    if (spanCache == null) {
+      spanCache = new SpanCache();
```

Comment:
This will work fine in most scenarios, but the correct singleton implementation would require double locking: https://www.baeldung.com/java-singleton-double-checked-locking

We can actually avoid this lazy initialization altogether and just declare
```
private static final SpanCache SPAN_CACHE = new SpanCache();
```
above and just return it here.